rust/worker/src/execution/orchestration/hnsw.rs

```diff
@@ -372,23 +376,94 @@ impl HnswQueryOrchestrator {
         }
     }
 
-    async fn merge_results_for_index(
-        &mut self,
-        ctx: &ComponentContext<Self>,
-        query_vector_index: usize,
-    ) {
+    async fn prefetch_record_data(&mut self, ctx: &ComponentContext<Self>, offset_ids: Vec<u32>) {
         let record_segment = self
             .record_segment
             .as_ref()
             .expect("Invariant violation. Record Segment is not set");
+        // TODO: Divide this into multiple tasks based on some criteria.
+        let offsetid_to_data_keys =
+            RecordSegmentKeys::OffsetIdToDataKeys(OffsetIdToDataKeys { keys: offset_ids });
+        let record_segment_info = RecordSegmentInfo {
+            keys: offsetid_to_data_keys,
+            segment: record_segment.clone(),
+            provider: self.blockfile_provider.clone(),
+        };
+        // TODO: Tag IO.
```

Comment:
i think the "tagging" should be part of the Operator, not the responsibility of the caller

Reply:
For sure, the next PR has tags in the operator. I did not really think a lot when writing this TODO here. It was mostly as a note to myself to not forget.